File: 771-jewels-and-stones/771-jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        int res = 0;
        unordered_set<char> hash(jewels.begin(), jewels.end());
        for (char s : stones)
        {
            if (hash.count(s)) res++;
        }
        return res;
    }   
    
};

File: 771-jewels-and-stones/771-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        int[] jewel = new int[256];
        for (char c: jewels.toCharArray()) {
            jewel[c]++;
        }
        
        int res = 0;
        for (char c:  stones.toCharArray()) {
            res += jewel[c];
        }
        
        return res;
    }
}